packages/payload/src/fields/config/types.ts

```diff
@@ -1330,6 +1330,46 @@ export type JoinFieldClient = {
 } & FieldBaseClient &
   Pick<JoinField, 'collection' | 'index' | 'maxDepth' | 'on' | 'type' | 'where'>
 
+export type FlattenBlock = {
+  flattenFields: FlattenField[]
+} & Block
+
+export type FlattenBlocksField = {
+  blocks: FlattenBlock[]
+} & BlocksField
+
+export type FlattenGroupField = {
+  flattenFields: FlattenField[]
+} & GroupField
+
+export type FlattenArrayField = {
+  flattenFields: FlattenField[]
+} & ArrayField
+
+export type FlattenTabAsField = {
+  flattenFields: FlattenField[]
+} & MarkRequired<TabAsField, 'name'>
+
+export type FlattenField =
```

Comment:
The name for this type is slightly off for me which hurts readability a bit. I would name it `FlattenedField` and change that on these other types too. `FlattenedArrayField` is the result of `flattenFields()` the function.